gateway/ln-gateway/src/bin/cln_extension.rs

```diff
@@ -559,59 +564,117 @@ impl GatewayLightning for ClnRpcService {
             .ok_or_else(|| tonic::Status::internal("Pruned Invoice was not supplied"))?;
         let payment_hash = sha256::Hash::from_slice(&pruned_invoice.payment_hash)
             .map_err(|err| tonic::Status::internal(err.to_string()))?;
+        let destination =
+            PublicKey::from_slice(&pruned_invoice.destination).expect("Should parse public key");
 
-        // Use a `riskfactor` of 10, which is the default for lightning-pay
-        let route = self
-            .get_route(pruned_invoice.clone(), 10)
-            .await
-            .map_err(|err| tonic::Status::internal(err.to_string()))?;
+        let mut excluded_nodes = vec![];
 
-        // Verify `max_delay` is greater than the worst case timeout for the payment
-        // failure in blocks
-        let delay = route
-            .first()
-            .ok_or_else(|| {
-                tonic::Status::internal(format!(
-                    "Returned route did not have any hops for payment_hash: {payment_hash}"
-                ))
-            })?
-            .delay;
-        if max_delay < delay.into() {
-            return Err(tonic::Status::internal(format!("Worst case timeout for the payment is too long. max_delay: {max_delay} delay: {delay} payment_hash: {payment_hash}")));
-        }
+        let payment_future = async {
+            let mut route_attempt = 0;
 
-        // Verify the total fee is less than `max_fee_msat`
-        let first_hop_amount = route
-            .first()
-            .ok_or_else(|| {
-                tonic::Status::internal(format!(
-                    "Returned route did not have any hops for payment_hash: {payment_hash}"
-                ))
-            })?
-            .amount_msat;
-        let last_hop_amount = route
-            .last()
-            .ok_or_else(|| {
-                tonic::Status::internal(format!(
-                    "Returned route did not have any hops for payment_hash: {payment_hash}"
-                ))
-            })?
-            .amount_msat;
-        let fee = first_hop_amount - last_hop_amount;
-        if max_fee_msat < fee.msat() {
-            return Err(tonic::Status::internal(format!(
-                "Fee: {} for payment {payment_hash} is greater than max_fee_msat: {max_fee_msat}",
-                fee.msat()
-            )));
-        }
+            loop {
+                let route = self
+                    .get_route(
+                        pruned_invoice.clone(),
+                        ROUTE_RISK_FACTOR,
+                        excluded_nodes.clone(),
+                    )
```

Comment:
will give us routes we can't use since it doesn't take max_delay

Reply:
Should we retry a new route in this case? Currently we just fail

Reply:
This isn't a strong enough reason to block anything, let's see if it's a problem in practice, I doubt it.